merlin/models/tf/transforms/features.py

```diff
@@ -104,23 +105,282 @@ class ToDense(FeaturesTensorTypeConversion):
+                            if f"{name}__values" not in inputs or f"{name}__offsets" not in inputs:
+                                raise ValueError(
+                                    f"The ragged list feature '{name}' is expected to be "
+                                    f"represented by two features in the inputs: '{name}__values' "
+                                    f"and '{name}__offsets', but they were not found."
+                                )
+                            val = list_col_to_ragged(
+                                inputs[f"{name}__values"], inputs[f"{name}__offsets"]
+                            )
+                            del inputs[f"{name}__values"]
+                            del inputs[f"{name}__offsets"]
+                        else:
+                            raise ValueError(f"Feature '{name}' was not found in the inputs")
+
+                    if len(val.shape) == 2:
+                        val = tf.expand_dims(val, axis=-1)
+
+                    if self.list_as_dense:
+                        val = to_dense(val, max_seq_length=col_schema_shape.dims[1].max)
+
+                elif name in inputs:
+                    val = inputs[name]
+
+                if val is not None:
+                    outputs[name] = val
+
+        # Adding other inputs that might not be in the schema,
+        # as they might be treated by other block
+        for k, v in inputs.items():
+            if k not in outputs:
+                outputs[k] = v
+
+        return outputs
+
+    def compute_output_shape(self, input_shapes):
+        if not input_shapes:
+            return input_shapes
+
+        output_shapes = {}
+        if self.has_schema:
+            for name in self.schema.column_names:
+                col_schema_shape = self.schema[name].shape
+                if col_schema_shape.is_list:
+                    if name not in input_shapes and f"{name}__offsets" in input_shapes:
+                        batch_size = input_shapes[f"{name}__offsets"][0]
+                        if batch_size is not None:
+                            # The length of offset is always batch size + 1
+                            batch_size -= 1
+
+                        seq_length = None
+                        if self.list_as_dense or not self.schema[name].shape.is_ragged:
+                            if col_schema_shape.dims[1] is None:
+                                raise Exception(
+                                    f"List feature {name} does not have maximum"
+                                    "length set in the schema ({col_schema_shape})"
+                                )
+                            seq_length = int(col_schema_shape.dims[1].max)
+
+                        output_shapes[name] = tf.TensorShape([batch_size, seq_length, 1])
+
+                    else:
+                        if len(input_shapes[name]) == 2:
+                            output_shapes[name] = input_shapes[name] + [1]
```

Comment:
Based on the error message from the test failures about how only tuples and not lists can be concatenated onto tuples, I think this might need to be:
```suggestion
                            output_shapes[name] = input_shapes[name] + (1,)
```

Reply:
Hey @karlhigley . Thanks for the review and suggestion. Typically we can concatenate `tf.TensorShape()` (`input_shapes[name]`) with `list`. I think CI is using the dataloader from main branch, which still returns tuple for list features.
I suspect for some reason the changes I did in [tox.ini](https://github.com/NVIDIA-Merlin/models/pull/999/commits/83f195334578e8fb5e0c7d0c04fe50882d7ca118) are not being able to install the dataloader from [PR #101](https://github.com/NVIDIA-Merlin/dataloader/pull/101).

Reply:
Curious: Does `TensorShape` concatenate with tuples too, or only lists?

Reply:
It looks like either one of these should work with `TensorShape` but tuples will only concat other tuples, so I think it's safe to switch over to a tuple here:
```
>>> shape = tf.TensorShape((1,2,3))
>>> shape + [4]
TensorShape([1, 2, 3, 4])
>>> shape + (4,)
TensorShape([1, 2, 3, 4])

>>> shape = (1,2,3)
>>> shape + [4]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate tuple (not "list") to tuple
>>> shape + (4,)
(1, 2, 3, 4)
```

Reply:
Sure @karlhigley . Changed it to tuple as you suggested